sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -105,7 +105,11 @@ def create_experiment(
             raise TimeoutError(
                 f"Timeout to create Katib Experiment: {namespace}/{experiment.metadata.name}"
             )
-        except Exception:
+        except Exception as e:
+            if hasattr(e, "status") and e.status == 409:
+                raise Exception(
+                    f"A Katib Experiment with the name {namespace}/{experiment.metadata.name} already exists."
+                )
```

Comment:
I think instead of checking the error, we should check if the requested Experiment already exists in the cluster before trying to create the Experiment in L97-L103.

cc: @andreyvelich

Reply:
I think, the downside of this that we require to call Kubernetes API server twice to always get Katib Experiment before creating this.
Would it be performance issue ?
On the other hand, KFP checks if Experiment exists before creating this: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L473-L475.

What are your thoughts @tenzen-y @droctothorpe @johnugeorge ?

Reply:
> Would it be performance issue ?

It might be better to improve the performance as much possible as. Although, I think the performance wouldn't be much worse even if we call the api-server twice.

> On the other hand, KFP checks if Experiment exists before creating this: https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py#L473-L475.

I'm ok with the way If we adopt a way, the same as KFP since I want to avoid depending status code.

Reply:
Addressed your feedback, @tenzen-y. Apologies for the delay.

Good point, @andreyvelich. FWIW, the performance impact is negligible, `tune` isn't called repeatedly in a way where performance is a concern, IMO the code is a bit more clean and clear with the explicit check, and aligning with the KFP SDK is a nice plus.

I could go either way and am happy to implement either pattern.

Reply:
I decoupled the commits just in case we want to revert to the exception handling approach 🙃 .

Reply:
@droctothorpe @tenzen-y What kind of pros and cons we have between:
- Calling API Server before creating Experiment
- Checking the Error code ?

Reply:
#### `list_experiment` first
- PRO: Cleaner code.
- PRO: Aligned with the KFP SDK.
- CON: Extra API call every time `tune` is called.

#### Handle 409
- PRO: [EAFP](https://docs.python.org/2/glossary.html#term-eafp).
- PRO: Less API calls.

Reply:
@droctothorpe @tenzen-y If we know that 409 code always represents that CR is already created do we have any other concerns why we don't want to follow this logic to check error code ?

For example, in the future we can handle other error codes (e.g. 401 means that user is not authorized to perform API call).

Reply:
Good point! Pushed up a revert commit. Happy to squash all 3 or they can be squashed at merge. Awaiting further input.

Reply:
@tenzen-y @johnugeorge What do you think about it ?

Reply:
Bump 🙃